File: tests/errors/defer.swift
public enum BasicError: Error {
	case requested
}

public func attempt(shouldThrow: Bool) throws -> Bool {
	if shouldThrow {
		throw BasicError.requested
	}
	return true
}

var processing: Bool

public func defers(shouldThrow: Bool) -> Bool? {
	processing = true
	defer {
		processing = false
	}
	return try? attempt(shouldThrow: shouldThrow)
}

File: tests/errors/defer.js
export function attempt$shouldThrow$(shouldThrow) {
  if (shouldThrow) {
    throw 0;
  }

  return true;
}
export function defers$shouldThrow$(shouldThrow) {
  try {
    processing[0] = true;
    let $try;

    try {
      $try = attempt$shouldThrow$(shouldThrow);
    } catch (e) {
      $try = null;
    }

    return $try;
  } finally {
    processing[0] = false;
  }
}
const processing = [false];